src/Sulu/Bundle/MediaBundle/Media/Storage/StorageInterface.php

```diff
@@ -22,29 +22,37 @@
      *
      * @param $tempPath
      * @param $fileName
-     * @param $version
+     * @param $preferredStorageOptions
      *
      * @return mixed
      */
-    public function save($tempPath, $fileName, $version);
+    public function save($tempPath, $fileName, $preferredStorageOptions = null);
 
     /**
-     * Give back the path to the document.
+     * Will return a file resource.
      *
-     * @param $fileName
-     * @param $version
-     * @param $storageOption
+     * @param $storageOptions
      *
-     * @return mixed
+     * @return resource
      */
-    public function load($fileName, $version, $storageOption);
+    public function load($storageOptions);
```

Comment:
don' t think that a save need to be consistency to a load. 

but as i look at the save method it could just be `save($tempPath, $preferedStorageOptions = null)` and to save the file with a correct filename (to make it easier to find the file) it would maybe good to send the filename as optional parameter `save($tempPath, $preferedStorageOptions = null, $fileName = null)` what you think @patkar.